test/DynamoCoreWpfTests/SplashScreenTests.cs

```diff
@@ -1,13 +1,34 @@
 
 
+using DynamoCoreWpfTests.Utility;
 using DynamoUtilities;
 using NUnit.Framework;
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace DynamoCoreWpfTests
 {
     [TestFixture]
     internal class SplashScreenTests
     {
+        public enum WindowsMessage
+        {
+            WM_CLOSE = 0x0010
+        }
+
+        [DllImport("user32.dll")]
+        public static extern IntPtr FindWindow(string lpClassName, String lpWindowName);
```

Comment:
Any reason you can't use WPF tree for this?

Reply:
The issue does not reproduce if the window is closed from managed code. THere are different code paths taken